docs/iommu.md

```diff
@@ -0,0 +1,87 @@
+# Virtual IOMMU
+
+## Rationales
+
+Having the possibility to expose a virtual IOMMU to the guest can be
+interesting to leverage for some use cases.
+
+The first reason why one might want to expose a virtual IOMMU to the
+guest is to increase the security regarding the memory accesses performed
+by the virtual devices, on behalf of the guest drivers. With a virtual
+IOMMU, the VMM can use interposition to make choices about letting a device
+access the guest memory, the same way a physical IOMMU would prevent a
+physical device from accessing the physical memory. The security gain is only
+visible from a guest perspective, which does not apply if the guest we're
+taking as example is corrupted. The benefits about security can be seen in
+case of nested virtualization as it can protect the first layer guest from
+malicious guests running in L2. 
+
+As suggested by the above statement, having a virtual IOMMU also allows for
+nested device passthrough. Let's say the system has a physical IOMMU, it is
+now possible to expose a virtual IOMMU to the guest, which will be responsible
+for updating the physical DMA remapping table. This will happen through the
+VFIO framework, and will eventually allow for passing a physical device through
+multiple layers of virtualization.
+
```

Comment:
Another question that this documentation triggers: if I do a device passthrough with `--iommu`, I assume the guest driver will switch to using IOVA and implictly expect our virtual IOMMU to do the IOVA->PA translation. I see that this PR passes the VFIO CI tests, though: Any idea how that works?

Reply:
Because we have to choose which devices are attached to the vIOMMU. By default, a device is not attached to it, therefore the driver does not use IOVAs.